src/__tests__/e2e/proxied-redirects.spec.ts

```diff
@@ -0,0 +1,20 @@
+import { test, expect } from '@playwright/test'
+
+test('should respect redirects specified for proxied domains', async ({
+	page,
+	context,
+	baseURL,
+}) => {
+	await context.addCookies([
+		{
+			name: 'hc_dd_proxied_site',
+			value: 'www.test-domain.io',
+			url: baseURL,
+		},
+	])
+
+  // see redirect fixture at proxied-redirects/www.test-domain.io.redirects.js
+	await page.goto('/source')
+	const { pathname } = new URL(page.url())
+	expect(pathname).toBe('/')
```

Comment:
I'm not sure if there's value-add to this additional case, or if it's just bloat, but leaving it here just in case

```ts
	{
		await page.goto('/source')
		const { pathname } = new URL(page.url())
		expect(pathname).toBe('/')
	}

	{
		await page.goto('/other')
		const { pathname } = new URL(page.url())
		expect(pathname).toBe('/other')
	}
```